### DISCUSSION TITLE
Does this kernel support manual battery limit?

### DISCUSSION THREAD

**Discussion by YosypC** [linux-surface/linux-surface#1166] [Q&A] [answered]
My Pro 8's UEFI has the Battery Limiter but it's a hard coded 50%, which makes it unusable out of home. Windows has a battery protection that automatically enables in certain conditions (never worked for me) to 80%. Is there any program on Linux capable of enforcing, if not an arbitrary value, that 80%?

**Comment by qzed** [comment, marked as answer]
Unfortunately, this seems to be impossible. From what I can tell, all charging/battery related logic is implemented directly in the EC, so the kernel doesn't have that much access. The 80% is set automatically somehow and there only seems to be some command to reset it (so that it charges fully to 100%). Arbitrary values are completely impossible as far as I can tell.

**Comment by LeftHandRev** [comment]
Turning off battery limit is what fixed it for me. Thanks, @qzed - I was ripping my hair out digging through ACPI modules.
